lib/components/form/settings-selector-panel.js

```diff
@@ -41,75 +57,196 @@ class SettingsSelectorPanel extends Component {
 
   _setBikeOnly = () => { this._setSoloMode('BICYCLE') }
 
-  _addBikeModeToQueryModes = (bikeMode) => {
+  _setMicromobilityOnly = () => { this._setSoloMode('MICROMOBILITY') }
+
+  /**
+   * Replace own mode with new mode. The only mode will have already been set,
+   * so this toggles whether the own mode includes a rental.
+   */
+  _replaceOwnMode = (newMode, referenceOwnMode) => {
     const { queryModes, setQueryParam } = this.props
-    const nonBikeModes = queryModes.filter(m => !m.startsWith('BICYCLE'))
-    setQueryParam({ mode: [...nonBikeModes, bikeMode].join(',') })
+    const nonOwnModes = queryModes.filter(m => !m.startsWith(referenceOwnMode))
+    setQueryParam({ mode: [...nonOwnModes, newMode].join(',') })
+  }
+
+  _setOwnBike = () => this._replaceOwnMode('BICYCLE', 'BICYCLE')
+
+  _setRentedBike = () => this._replaceOwnMode('BICYCLE_RENT', 'BICYCLE')
+
+  _setOwnVehicle = () => this._replaceOwnMode('MICROMOBILITY', 'MICROMOBILITY')
+
+  _setRentedVehicle = () => {
```

Comment:
Again, the use of the generic vehicle language feels like a bit of a mismatch.

Reply:
Will add comments